blockchain/blockchain_test.go

```diff
@@ -724,3 +724,55 @@ func TestSubscribeL1Head(t *testing.T) {
 	require.True(t, ok)
 	assert.Equal(t, l1Head, got)
 }
+
+func fetchStateUpdatesAndBlocks(samples int) ([]*core.StateUpdate, []*core.Block, error) {
+	client := feeder.NewClient(utils.Mainnet.FeederURL)
+	gw := adaptfeeder.New(client)
+
+	suList := make([]*core.StateUpdate, samples)
+	blocks := make([]*core.Block, samples)
+	for i := range samples {
+		fmt.Println("fetching", i)
+		su, err := gw.StateUpdate(context.Background(), uint64(i))
+		if err != nil {
+			return nil, nil, err
+		}
+		suList[i] = su
+		block, err := gw.BlockByNumber(context.Background(), uint64(i))
+		if err != nil {
+			return nil, nil, err
+		}
+		blocks[i] = block
+	}
+	return suList, blocks, nil
+}
+
+func BenchmarkBlockchainStore(b *testing.B) {
+	samples := 100
+	stateUpdates, blocks, err := fetchStateUpdatesAndBlocks(samples)
+	require.NoError(b, err)
+
+	b.Run("new", func(b *testing.B) {
+		for b.Loop() {
+			b.StopTimer()
+			chain := blockchain.New(memory.New(), &utils.Mainnet, true)
+			b.StartTimer()
+
+			for j := range samples {
+				require.NoError(b, chain.Store(blocks[j], &emptyCommitments, stateUpdates[j], nil))
+			}
+		}
+	})
+
+	b.Run("old", func(b *testing.B) {
+		for b.Loop() {
+			b.StopTimer()
+			chain := blockchain.New(memory.New(), &utils.Mainnet, false)
+			b.StartTimer()
+
+			for j := range samples {
+				require.NoError(b, chain.Store(blocks[j], &emptyCommitments, stateUpdates[j], nil))
+			}
+		}
+	})
```

Comment:
We should write a for loop for `useNewState`. It's not only about less code, but also about guarantee that the structure of the 2 tests are the same.